mmgen/models/architectures/DDPM/modules.py

```diff
@@ -0,0 +1,324 @@
+from copy import deepcopy
+from functools import partial
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import ACTIVATION_LAYERS
+from mmcv.cnn.bricks import build_activation_layer, build_norm_layer
+from mmcv.cnn.utils import constant_init
+
+from mmgen.models.builder import MODULES, build_module
+
+
+class EmbedSequential(nn.Sequential):
+    """A sequential module that passes timestep embeddings to the children that
+    support it as an extra input.
+
+    Modified from
+    https://github.com/openai/improved-diffusion/blob/main/improved_diffusion/unet.py#L35
+    """
+
+    def forward(self, x, y):
+        for layer in self:
+            if isinstance(layer, DenoisingResBlock):
+                x = layer(x, y)
+            else:
+                x = layer(x)
+        return x
+
+
+@ACTIVATION_LAYERS.register_module()
+class SiLU(nn.Module):
+
+    def __init__(self, inplace=False):
+        super().__init__()
+        self.inplace = inplace
+
+    def forward(self, x):
+        # return F.silu(x, inplace=self.inplace)
+        return torch.sigmoid(x) * x
```

Comment:
Dose `inplace` make sense here?

Reply:
I forget to add `TODO` to this comment.
Official code realizes `SiLU` as L:41, if we use torch's official implementation, we may have an error of 1e-5.
I'm not sure which version to use.